### PR TITLE
Remove unnecessary GitHub Token from checkout step in GitHub Actions workflows

### LINKED ISSUE #1390: Remove unncessary GitHub Token from checkout step in GitHub Actions workflows
We use `GITHUB_TOKEN` in the checkout steps in many GitHub Action Workflows

https://github.com/vmware-tanzu/tce/blob/main/.github/workflows/release-ga.yaml#L29-L32

https://github.com/vmware-tanzu/tce/blob/main/.github/workflows/release-fake.yaml#L28-L31

https://github.com/vmware-tanzu/tce/blob/main/.github/workflows/release-nonga.yaml#L30-L33

But the token isn't really necessary and isn't used, so it's better to remove them as it's kind of like unused code / dead code

### PR DESCRIPTION
**Pull request by karuppiah7890**
## What this PR does / why we need it

The token isn't really necessary and isn't used, so it's better to remove
them as it's kind of like unused code / dead code

Fixes #1390

## Details for the Release Notes

```release-note
NONE
```

## Which issue(s) this PR fixes

Fixes: #1390 

## Describe testing done for PR

None

## Special notes for your reviewer

None

### CODE REVIEW

**Review by navidshaikh — APPROVED** [review]
lgtm

**Review by stmcginnis — APPROVED** [review]
Appears to be fine.

**Review by davidvonthenen — COMMENTED** [review]
If I remember correctly, I thought the repo had to be checked out using a token in order for the repo to have write access. We can always try and revert if it doesn't work. It would have been nice to keep the repo workflows static before renaming then merging this changes.

**Pull request merged**
